### ISSUE TITLE
Potential issue with implementation

### ISSUE THREAD

**Issue by Omig12** [tensorflow/addons#2738]
I think some of the operations might be out of order in the following lines: 

https://github.com/tensorflow/addons/blob/b2dafcfa74c5de268b8a5c53813bc0b89cadf386/tensorflow_addons/losses/contrastive.py#L67-L69

I'm basing this on a comparison with the implementation of contrastive loss on: https://keras.io/examples/vision/siamese_contrastive/

```py
def contrastive_loss(y_true, y_pred):
        """Calculates the constrastive loss.

        Arguments:
            y_true: List of labels, each label is of type float32.
            y_pred: List of predictions of same length as of y_true,
                    each label is of type float32.

        Returns:
            A tensor containing constrastive loss as floating point value.
        """

        square_pred = tf.math.square(y_pred)
        margin_square = tf.math.square(tf.math.maximum(margin - (y_pred), 0))
        return tf.math.reduce_mean(
            (1 - y_true) * square_pred + (y_true) * margin_square
        )
```

**Comment by bhack** [comment]
Have you tested if this is giving different results from the Tutorial loss or from https://github.com/tensorflow/models/blob/master/official/vision/beta/projects/simclr/losses/contrastive_losses.py#L60 ?

As we are having many contrastive loss duplication between official tutorials, Addons, model gardens etc. (/cc @theadactyl) what do you think about porting/contributing one in [Keras-cv](https://github.com/keras-team/keras-cv) directly?

**Comment by Omig12** [comment]
I tested it by passing the following vectors: 

No Loss:
`y_true = [1, 0, 1]` and `y_pred = [1, 0, 1]` 

Small Loss:
`y_true = [1, 0, 1]` and `y_pred = [1, 1, 1]` 

Total Loss: 
`y_true = [1, 0, 1]` and `y_pred = [0, 1, 0]`
